### PR TITLE
Refactor docker-compose dev files and env vars

### LINKED ISSUE #896: Easier docker-compose start script
- [x] The feature was discussed in [discussions](https://github.com/openkfw/TruBudget/discussions) of this repository.
- [x] I have searched the [issues](https://github.com/openkfw/TruBudget/issues) of this repository and believe that this is not a duplicate.

## Description 💡

Currently it's difficult to quickly setup a developer environment without carring to much about all configurations like which env vars shiould i use which services should i comment out or which docker-compose file should i use for my usecase.
The goal should be one entry point where I can decide via arguments what environment I need to setup for my specific usecase.
So a well documented script which starts the environment of Trubudget in one command where I can easily specifiy my needs as developer.

### TODOs
- [ ] create dev start script with following features:
         - choosable services (include/exclude) 
         - choose which service should be hotreloaded
         - choose if data should be persisted
         - choose if provisioning is needed
         - useful help section which is printed if the script was executed in a wrong way
         - think of useful features
- [x] create docker-compose where all services are included
- [ ] Check all available docker-compose files and try to minimize the amount of docker-compose files

### PR DESCRIPTION
**Pull request by daniel-arnauer**
- [x] Refactor docker-compose file for development (#896) 
- [x] Refactor env-files for better overview

How to test:

- go to root directory of TruBudget
- slim setup (blockchain, api, frontend, provisioning):  `sh scripts/development/start-dev.sh --slim`
- full setup (all entites): `sh scripts/development/start-dev.sh --full` 

Closes #896

### CODE REVIEW

File: storage-service/.env.example [hunk 1]
```diff
@@ -11,7 +11,6 @@ ROOT_SECRET=root-secret
 API_HOST=localhost
 API_PORT=8080
 ENVIRONMENT_TYPE=TEST
-STORAGE_SERVICE_ENABLED="true"
 STORAGE_SERVICE_EXTERNAL_URL=localhost
```
Comment (daniel-arnauer) [inline storage-service/.env.example:14]: dead env var

File: docs/developer/developer-setup.md [hunk 2]
```diff
@@ -169,26 +169,34 @@ If you want to start developing on Trubudget, you need to setup the application
...
+- Excel export
+- Email notification
+- Email Database
+- Storage Service
+- Minio Database (external document storage)
 
-It takes some time to build and run at the first launch. After that, source codes of API and frontend are live reloaded. That means any change in `./src` folder is reflected in respective containter automatically.
+If the script is started with `--slim`, only the Blockchain, API, Provisioning and Frontend will be started.
+
+It takes some time to build and run at the first launch. After that, source codes of all services are live reloaded. That means any change in `./src` folder is reflected in respective container automatically.
```
Comment (Stezido) [inline docs/developer/developer-setup.md:190]: I would expect all services to run without hot reloading per default

File: docs/developer/developer-setup.md [hunk 3]
```diff
@@ -169,26 +169,34 @@ If you want to start developing on Trubudget, you need to setup the application
...
 
 The frontend should be availaible as usual at http://localhost:3000
 
 Docker Compose ensures that services are communicating and have correct environment variables set. Docker Compose puts all services in the same network and exposes needed ports.
 
 You can inspect each container individually: `docker logs --follow CONTAINER`. Where _CONTAINER_ represents selected value of NAMES column container in output of `docker ps` command.
 
+Note: Hot reloading with docker requires a lot of processing power (due to docker volumes). If you experience huge delays while hot-reloading, you should start the services you need by yourself without docker.
```
Comment (Stezido) [inline docs/developer/developer-setup.md:198]: as described above I would expect the services to run without hot reloading per default

File: docs/developer/developer-setup.md [hunk 4]
```diff
@@ -169,26 +169,34 @@ If you want to start developing on Trubudget, you need to setup the application
...
 - Blockchain (master node)
 - API (master API)
 - Provisioning (feeds application with dummy data)
-- Excel export
 - Frontend
+- Excel export
+- Email notification
+- Email Database
+- Storage Service
+- Minio Database (external document storage)
```
Comment (Stezido) [inline docs/developer/developer-setup.md:187]: suggestion to add to the doc:
For further details how to use the script print out its help section with 
```
sh scripts/development/start-dev.sh -h
```

File: scripts/development/start-dev.sh [hunk 5]
```diff
@@ -1,13 +1,105 @@
...
+    echo
+    echo "Hint: Make sure you are in the root directory of TruBudget."
+    echo
+    echo "Syntax: sh scripts/development/start-dev.sh [options]"
+    echo "Example: sh scripts/development/start-dev.sh --full --build"
+    echo
+    echo "options:"
+    echo "      --slim                  Starts a TruBudget instance with master-node, master-api, provisioning and frontend."
+    echo "      --full                  Starts a TruBudget instance with master-node, emaildb, minio, master-api, email-service,"
+    echo "                              provisioning, excel-export-service, storage and frontend."
```
Comment (Stezido) [inline scripts/development/start-dev.sh:24]: extend the info by the default behaviour. I don't know what the script is doing without an option- would expect Trubudget without any service enabled

File: scripts/development/start-dev.sh [hunk 6]
```diff
@@ -1,13 +1,105 @@
...
+    echo "      --build                 Force docker-compose built"
+    echo "-h  | --help                  Print the help section"
+    echo
```
Comment (Stezido) [inline scripts/development/start-dev.sh:27]: What's about this script feedback?:
without any option only blockchain/api/frontend shall start
--provision (start provisioning service)
--enable-service export (source additional env vars & start export service)
--enable-service email (source additional env vars & start email service)
--enable-service storage (source additional env vars & start stoprage service and minio)
--full as above
--build as above
--debug [service] (enable hotreloading for the service specified)
-h --help as above

**Review by Stezido — COMMENTED** [review]
I wouldn't set the log output color to yellow/orange since this color is mostly indicating a warning.
I suggest prefixing the info outputs with `INFO:` 
And the warnings with `WARNING:` they can be yellow/orange

File: docs/developer/developer-setup.md (hunk 3 above)
Comment (mayrmartin) [reply docs/developer/developer-setup.md:198]: @daniel-arnauer consider using admonitions (https://docusaurus.io/docs/next/markdown-features/admonitions) for notes,tips and info

File: .gitignore [hunk 7]
```diff
@@ -11,3 +11,4 @@ yarn-error.log
 *.code-workspace
 .vs/
 e2e-test/debug.log
+minio
```
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply .gitignore:14]: Because the auto generated files are annoying

File: scripts/development/start-dev.sh (hunk 5 above)
Comment (daniel-arnauer) [reply scripts/development/start-dev.sh:24]: Without any option, I would expect a full instance

File: frontend/.env [hunk 8]
```diff
@@ -8,3 +8,11 @@ RPC_PASSWORD=s750SiJnj50yIrmwxPnEdSzpfGlTAHzhaUwgqKeb0G1j
 API_HOST=api
 API_PORT=8080
 ENVIRONMENT_TYPE=TEST
+# set USE_SSL=ssl for https or USE_SSL= for http
```
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply frontend/.env:11]: done

File: provisioning/src/index.js [hunk 9]
```diff
@@ -69,6 +74,8 @@ const impersonate = async (userId, password) => {
   const token = await authenticate(axios, userId, password);
   console.log(`Now logged in as ${userId}`);
   axios.defaults.headers.common.Authorization = `Bearer ${token}`;
+  currentUser.userId = userId;
```
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply provisioning/src/index.js]: done

File: scripts/development/start-dev.sh (hunk 5 above)
Comment (daniel-arnauer) [reply scripts/development/start-dev.sh:24]: done - added:"Default option: full TruBudget instance with all services in debug mode"

File: docs/developer/developer-setup.md (hunk 4 above)
Comment (daniel-arnauer) [reply docs/developer/developer-setup.md:187]: done

File: scripts/development/start-dev.sh (hunk 6 above)
Comment (daniel-arnauer) [reply scripts/development/start-dev.sh:27]: for tgis feedback, I created a follow up issue: #904

File: docs/developer/developer-setup.md (hunk 3 above)
Comment (daniel-arnauer) [reply docs/developer/developer-setup.md:198]: done

**Comment by daniel-arnauer** [comment]
follow up issue: #904

File: docs/developer/developer-setup.md (hunk 2 above)
Comment (Stezido) [reply docs/developer/developer-setup.md:191]: Will be resolved in the follow up issue #904

File: docs/developer/developer-setup.md (hunk 3 above)
Comment (Stezido) [reply docs/developer/developer-setup.md]: the hot reloading will be adapted in #904

File: .gitignore (hunk 7 above)
[the remark this answers is not in the document]
Comment (daniel-arnauer) [reply .gitignore]: done - removed

**Review by Stezido — APPROVED** [review]

**Pull request merged**